[BUG-Flashing] does not compile on ESPHome 2025.7.3

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Is this question already covered by the FAQ?

- [x] I have read the FAQ.

### Are you trying to run ESPHome on another Dreo Fan than the DR-HTF004S?

This is a DR-HTF004S.

### Your Fan Model (ONLY the DR-HTF004S is supported at the moment).

n/a

### Your Fan Firmware Version.

ESPHome 2025.6.x

### Your Fan MCU Version.

do not know

### Current Issue Flashing

Since ESPHome 2025.7 compilation is no longer successful. There was a bug in ESPHome 2025.7 wrt. string lambdas that seems to be fixed now (https://github.com/esphome/esphome/pull/9543), however something seems still broken. 
The corresponding issue I found is not planned to be fixed: https://github.com/esphome/esphome/issues/9625 

Thanks in advance for giving it a look!
Andreas


Compilation log below:

```
INFO ESPHome 2025.7.3
INFO Reading configuration /config/esphome/dreo-pilot-max-s-fan-dr-htf004s.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing dreo-pilot-max-s-fan-dr-htf004s (board: generic-bk7231n-qfn32-tuya; framework: arduino; platform: libretiny)
--------------------------------------------------------------------------------
Library Manager: Installing ESP32Async/AsyncTCP @ 3.4.5
INFO Installing ESP32Async/AsyncTCP @ 3.4.5
Unpacking  [####################################]  100%
Library Manager: AsyncTCP@3.4.5 has been installed!
INFO AsyncTCP@3.4.5 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
INFO Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.7.10 has been installed!
INFO ESPAsyncWebServer@3.7.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/noise-c @ 0.1.10
INFO Installing esphome/noise-c @ 0.1.10
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.10 has been installed!
INFO noise-c@0.1.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10020.7
INFO Installing esphome/libsodium @ 1.10020.7
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10020.7 has been installed!
INFO libsodium@1.10020.7 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 7.4.2
INFO Installing bblanchon/ArduinoJson @ 7.4.2
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@7.4.2 has been installed!
INFO ArduinoJson@7.4.2 has been installed!
Warning! Non-Git installations are NOT SUPPORTED.
HARDWARE: BK7231N 120MHz, 256KB RAM, 1.03MB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a 
 - framework-beken-bdk @ 0.0.0+v2021.06.07.sha.6491b8c 
 - library-flashdb @ 1.2.0+sha.d5c892f 
 - library-freertos @ 9.0.0+sha.ff31b7d 
 - library-freertos-port @ 2025.1.18+sha.dbfc408 
 - library-lwip @ 2.2.0-bdk+sha.48da4a2 
 - library-printf @ 6.1.0+sha.28a79bd 
 - library-uf2ota @ 5.0.0+sha.f955412 
PLATFORM VERSIONS:
 - libretiny @ 1.9.1
 - ltchiptool @ 4.12.1
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.7.3
Library Manager: Installing DNSServer
INFO:Library Manager:Installing DNSServer
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Library Manager: DNSServer@1.1.0 has been installed!
INFO:Library Manager:DNSServer@1.1.0 has been installed!
Dependency Graph
|-- AsyncTCP @ 3.4.5
|-- ESPAsyncWebServer @ 3.7.10
|-- DNSServer @ 1.1.0
|-- noise-c @ 0.1.10
|-- ArduinoJson @ 7.4.2
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/esphome/ota/ota_esphome.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/json/json_util.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/libretiny/core.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/libretiny/gpio_arduino.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/libretiny/helpers.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/libretiny/lt_component.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/libretiny/preferences.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/logger/logger_libretiny.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/mdns/mdns_libretiny.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/ota/ota_backend_arduino_libretiny.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/select/select.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/select/select_call.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/select/select_traits.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/sensor/sensor.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/socket/lwip_sockets_impl.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/socket/socket.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/switch/automation.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/switch/switch.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/template/select/template_select.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/template/sensor/template_sensor.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/template/switch/template_switch.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/uart/switch/uart_switch.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/uart/uart.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/uart/uart_component.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/uart/uart_component_libretiny.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/uart/uart_debugger.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/web_server/list_entities.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/web_server/ota/ota_web_server.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/web_server/web_server.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/web_server/web_server_v1.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/components/wifi/wifi_component_libretiny.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/application.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/color.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/component.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/controller.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/log.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/string_ref.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/time.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/esphome/core/util.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/main.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libf7d/AsyncTCP/AsyncTCP.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncEventSource.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncJson.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncMessagePack.cpp.o
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncWebHeader.cpp.o
Archiving .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libf7d/libAsyncTCP.a
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncWebServerRequest.cpp.o
/config/esphome/dreo-pilot-max-s-fan-dr-htf004s.yaml: In lambda function:
/config/esphome/dreo-pilot-max-s-fan-dr-htf004s.yaml:518:72: error: no matching function for call to 'to_string(std::__cxx11::basic_string<char>&)'
  518 |           id(fan_mode).publish_state(to_string(id(fan_mode_memory_global)).c_str());
      |                                                                        ^
In file included from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/string:55,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/locale_classes.h:40,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ios_base.h:41,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/streambuf:41,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/streambuf_iterator.h:35,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/iterator:66,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ranges_algobase.h:36,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ranges_algo.h:35,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/algorithm:64,
                 from /data/cache/platformio/platforms/libretiny/cores/common/arduino/src/Arduino.h:14,
                 from <command-line>:
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6688:3: note: candidate: 'std::string std::__cxx11::to_string(long double)'
 6688 |   to_string(long double __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6688:25: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long double'
 6688 |   to_string(long double __val)
      |             ~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6679:3: note: candidate: 'std::string std::__cxx11::to_string(double)'
 6679 |   to_string(double __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6679:20: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'double'
 6679 |   to_string(double __val)
      |             ~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6670:3: note: candidate: 'std::string std::__cxx11::to_string(float)'
 6670 |   to_string(float __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6670:19: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'float'
 6670 |   to_string(float __val)
      |             ~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6659:3: note: candidate: 'std::string std::__cxx11::to_string(long long unsigned int)'
 6659 |   to_string(unsigned long long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6659:32: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long long unsigned int'
 6659 |   to_string(unsigned long long __val)
      |             ~~~~~~~~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6647:3: note: candidate: 'std::string std::__cxx11::to_string(long long int)'
 6647 |   to_string(long long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6647:23: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long long int'
 6647 |   to_string(long long __val)
      |             ~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6639:3: note: candidate: 'std::string std::__cxx11::to_string(long unsigned int)'
 6639 |   to_string(unsigned long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6639:27: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long unsigned int'
 6639 |   to_string(unsigned long __val)
      |             ~~~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6628:3: note: candidate: 'std::string std::__cxx11::to_string(long int)'
 6628 |   to_string(long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6628:18: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long int'
 6628 |   to_string(long __val)
      |             ~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6620:3: note: candidate: 'std::string std::__cxx11::to_string(unsigned int)'
 6620 |   to_string(unsigned __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6620:22: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'unsigned int'
 6620 |   to_string(unsigned __val)
      |             ~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6609:3: note: candidate: 'std::string std::__cxx11::to_string(int)'
 6609 |   to_string(int __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6609:17: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'int'
 6609 |   to_string(int __val)
      |             ~~~~^~~~~
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/AsyncWebSocket.cpp.o
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:154: warning: "htons" redefined
  154 | #define htons(x) __htons(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:119: note: this is the location of the previous definition
  119 | #define htons(x) lwip_htons(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:155: warning: "ntohs" redefined
  155 | #define ntohs(x) __ntohs(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:120: note: this is the location of the previous definition
  120 | #define ntohs(x) lwip_ntohs(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:156: warning: "htonl" redefined
  156 | #define htonl(x) __htonl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:121: note: this is the location of the previous definition
  121 | #define htonl(x) lwip_htonl(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:157: warning: "ntohl" redefined
  157 | #define ntohl(x) __ntohl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo-pilot-max-s-fan-dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:122: note: this is the location of the previous definition
  122 | #define ntohl(x) lwip_ntohl(x)
      | 
Compiling .pioenvs/dreo-pilot-max-s-fan-dr-htf004s/libb81/ESPAsyncWebServer/BackPort_SHA1Builder.cpp.o
*** [.pioenvs/dreo-pilot-max-s-fan-dr-htf004s/src/main.cpp.o] Error 1
========================= [FAILED] Took 268.64 seconds =========================
```

### Expected Behavior

Compilation and flashing an update also works with ESPHome 2025.7 and above.

### Extra information about the steps taken so far ⚠️ Please Respond to these.

```markdown
- Did you put your fan in "pairing mode" prior?
n/a, building the firmware fails before attempting to flash

- Did you try other endpoints (/mcu.html, /devinfoget, etc.)? (List all endpoints found)
n/a, building the firmware fails before attempting to flash

- Which other endpoints worked and what was the response you got? (Dump the full request/response)
n/a, building the firmware fails before attempting to flash
```

### Information about your browser environment at time of flashing.

- Browser: n/a, building the firmware fails before attempting to flash
- Version: n/a, building the firmware fails before attempting to flash

Same here:
```
INFO ESPHome 2025.9.1
INFO Reading configuration /config/esphome/dreo-space-heater.yaml...
WARNING 'dreo_dr-htf004s': Using the '_' (underscore) character in the hostname is discouraged as it can cause problems with some DHCP and local name services. For more information, see https://esphome.io/guides/faq.html#why-shouldn-t-i-use-underscores-in-my-device-name
INFO Generating C++ source...
INFO Compiling app...
Processing dreo_dr-htf004s (board: generic-bk7231n-qfn32-tuya; framework: arduino; platform: libretiny)
--------------------------------------------------------------------------------
Platform Manager: Installing libretiny
INFO Installing libretiny
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: libretiny@1.9.2 has been installed!
INFO libretiny@1.9.2 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/framework-beken-bdk#v2021.06.07
INFO Installing git+https://github.com/libretiny-eu/framework-beken-bdk#v2021.06.07
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-gqe7fjub'...
Note: switching to '6491b8c001b737c790cba0bee16e548859d5e6fe'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Updating files: 100% (3077/3077), done.
Tool Manager: framework-beken-bdk@0.0.0+v2021.06.07.sha.6491b8c has been installed!
INFO framework-beken-bdk@0.0.0+v2021.06.07.sha.6491b8c has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/ArduinoCore-API#2022.08.24
INFO Installing git+https://github.com/libretiny-eu/ArduinoCore-API#2022.08.24
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-6fyz8w0x'...
Note: switching to '237b10a4f6da14bf32e5e0963b7f28865216a60c'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: framework-arduino-api@2022.8.24+sha.237b10a has been installed!
INFO framework-arduino-api@2022.8.24+sha.237b10a has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/lwip#2.2.0-bdk
INFO Installing git+https://github.com/libretiny-eu/lwip#2.2.0-bdk
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-tafwhu9k'...
Tool Manager: library-lwip@2.2.0-bdk+sha.48da4a2 has been installed!
INFO library-lwip@2.2.0-bdk+sha.48da4a2 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-freertos#9.0.0
INFO Installing git+https://github.com/libretiny-eu/library-freertos#9.0.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-dajutro3'...
Tool Manager: library-freertos@9.0.0+sha.ff31b7d has been installed!
INFO library-freertos@9.0.0+sha.ff31b7d has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-freertos-port#2025.01.18
INFO Installing git+https://github.com/libretiny-eu/library-freertos-port#2025.01.18
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-qko4x2n0'...
Note: switching to 'dbfc408e47f57d3ca9a22e982faf0094c3cfc7a6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-freertos-port@2025.1.18+sha.dbfc408 has been installed!
INFO library-freertos-port@2025.1.18+sha.dbfc408 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-flashdb#1.2.0
INFO Installing git+https://github.com/libretiny-eu/library-flashdb#1.2.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-n_36lt8z'...
Note: switching to 'd5c892f0c545884e9f0de0cf67215e426dc8109d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-flashdb@1.2.0+sha.d5c892f has been installed!
INFO library-flashdb@1.2.0+sha.d5c892f has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-printf#6.1.0
INFO Installing git+https://github.com/libretiny-eu/library-printf#6.1.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-5oc_sfb8'...
Note: switching to '28a79bdd114da9c84880537dec2a1399d82823d3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-printf@6.1.0+sha.28a79bd has been installed!
INFO library-printf@6.1.0+sha.28a79bd has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-uf2ota#5.0.0
INFO Installing git+https://github.com/libretiny-eu/library-uf2ota#5.0.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-2lgg9cvh'...
Note: switching to 'f955412ed1b196fb2b09f4470ed54dcd9b026d03'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-uf2ota@5.0.0+sha.f955412 has been installed!
INFO library-uf2ota@5.0.0+sha.f955412 has been installed!
Tool Manager: Installing toolchain-gccarmnoneeabi @ ~1.100301.0
INFO Installing toolchain-gccarmnoneeabi @ ~1.100301.0
Downloading  [####################################]  100%          
Unpacking  [####################################]  100%          
Tool Manager: toolchain-gccarmnoneeabi@1.100301.220327 has been installed!
INFO toolchain-gccarmnoneeabi@1.100301.220327 has been installed!
Library Manager: Installing ESP32Async/AsyncTCP @ 3.4.5
INFO Installing ESP32Async/AsyncTCP @ 3.4.5
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: AsyncTCP@3.4.5 has been installed!
INFO AsyncTCP@3.4.5 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
INFO Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.7.10 has been installed!
INFO ESPAsyncWebServer@3.7.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/noise-c @ 0.1.10
INFO Installing esphome/noise-c @ 0.1.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.10 has been installed!
INFO noise-c@0.1.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10020.7
INFO Installing esphome/libsodium @ 1.10020.7
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10020.7 has been installed!
INFO libsodium@1.10020.7 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 7.4.2
INFO Installing bblanchon/ArduinoJson @ 7.4.2
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@7.4.2 has been installed!
INFO ArduinoJson@7.4.2 has been installed!
LibreTiny: Creating a virtual environment for Python dependencies
LibreTiny: Installing Python dependencies
Collecting ltchiptool<5.0,>=4.12.2
  Downloading ltchiptool-4.12.2-py3-none-any.whl.metadata (4.3 kB)
Collecting bitstruct<9.0.0,>=8.1.1 (from ltchiptool<5.0,>=4.12.2)
  Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (7.3 kB)
Collecting bk7231tools<3.0.0,>=2.0.0 (from ltchiptool<5.0,>=4.12.2)
  Downloading bk7231tools-2.1.0-py3-none-any.whl.metadata (841 bytes)
Collecting click<9.0.0,>=8.1.3 (from ltchiptool<5.0,>=4.12.2)
  Downloading click-8.3.0-py3-none-any.whl.metadata (2.6 kB)
Collecting colorama<0.5.0,>=0.4.5 (from ltchiptool<5.0,>=4.12.2)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting hexdump<4.0,>=3.3 (from ltchiptool<5.0,>=4.12.2)
  Downloading hexdump-3.3.zip (12 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting importlib-metadata (from ltchiptool<5.0,>=4.12.2)
  Downloading importlib_metadata-8.7.0-py3-none-any.whl.metadata (4.8 kB)
Collecting prettytable<4.0.0,>=3.3.0 (from ltchiptool<5.0,>=4.12.2)
  Downloading prettytable-3.16.0-py3-none-any.whl.metadata (33 kB)
Collecting py-datastruct<2.0.0,>=1.0.0 (from ltchiptool<5.0,>=4.12.2)
  Downloading py_datastruct-1.1.0-py3-none-any.whl.metadata (10 kB)
Collecting pyaes<2.0.0,>=1.6.1 (from ltchiptool<5.0,>=4.12.2)
  Downloading pyaes-1.6.1.tar.gz (28 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting requests<3.0.0,>=2.31.0 (from ltchiptool<5.0,>=4.12.2)
  Downloading requests-2.32.5-py3-none-any.whl.metadata (4.9 kB)
Collecting semantic-version<3.0.0,>=2.10.0 (from ltchiptool<5.0,>=4.12.2)
  Downloading semantic_version-2.10.0-py2.py3-none-any.whl.metadata (9.7 kB)
Collecting xmodem<0.5.0,>=0.4.6 (from ltchiptool<5.0,>=4.12.2)
  Downloading xmodem-0.4.7-py3-none-any.whl.metadata (4.3 kB)
Collecting ymodem<2.0.0,>=1.5.1 (from ltchiptool<5.0,>=4.12.2)
  Downloading ymodem-1.5.1-py3-none-any.whl.metadata (5.6 kB)
Collecting pyserial<4.0,>=3.5 (from bk7231tools<3.0.0,>=2.0.0->ltchiptool<5.0,>=4.12.2)
  Downloading pyserial-3.5-py2.py3-none-any.whl.metadata (1.6 kB)
Collecting wcwidth (from prettytable<4.0.0,>=3.3.0->ltchiptool<5.0,>=4.12.2)
  Downloading wcwidth-0.2.14-py2.py3-none-any.whl.metadata (15 kB)
Collecting charset_normalizer<4,>=2 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.2)
  Downloading charset_normalizer-3.4.3-cp312-cp312-manylinux2014_aarch64.manylinux_2_17_aarch64.manylinux_2_28_aarch64.whl.metadata (36 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.2)
  Downloading idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting urllib3<3,>=1.21.1 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.2)
  Downloading urllib3-2.5.0-py3-none-any.whl.metadata (6.5 kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.2)
  Downloading certifi-2025.8.3-py3-none-any.whl.metadata (2.4 kB)
Collecting ordered-set>=4.1.0 (from ymodem<2.0.0,>=1.5.1->ltchiptool<5.0,>=4.12.2)
  Downloading ordered_set-4.1.0-py3-none-any.whl.metadata (5.3 kB)
Collecting zipp>=3.20 (from importlib-metadata->ltchiptool<5.0,>=4.12.2)
  Downloading zipp-3.23.0-py3-none-any.whl.metadata (3.6 kB)
Downloading ltchiptool-4.12.2-py3-none-any.whl (301 kB)
Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (84 kB)
Downloading bk7231tools-2.1.0-py3-none-any.whl (37 kB)
Downloading click-8.3.0-py3-none-any.whl (107 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading prettytable-3.16.0-py3-none-any.whl (33 kB)
Downloading py_datastruct-1.1.0-py3-none-any.whl (28 kB)
Downloading requests-2.32.5-py3-none-any.whl (64 kB)
Downloading semantic_version-2.10.0-py2.py3-none-any.whl (15 kB)
Downloading xmodem-0.4.7-py3-none-any.whl (35 kB)
Downloading ymodem-1.5.1-py3-none-any.whl (21 kB)
Downloading importlib_metadata-8.7.0-py3-none-any.whl (27 kB)
Downloading certifi-2025.8.3-py3-none-any.whl (161 kB)
Downloading charset_normalizer-3.4.3-cp312-cp312-manylinux2014_aarch64.manylinux_2_17_aarch64.manylinux_2_28_aarch64.whl (146 kB)
Downloading idna-3.10-py3-none-any.whl (70 kB)
Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)
Downloading pyserial-3.5-py2.py3-none-any.whl (90 kB)
Downloading urllib3-2.5.0-py3-none-any.whl (129 kB)
Downloading zipp-3.23.0-py3-none-any.whl (10 kB)
Downloading wcwidth-0.2.14-py2.py3-none-any.whl (37 kB)
Building wheels for collected packages: hexdump, pyaes
  Building wheel for hexdump (pyproject.toml): started
  Building wheel for hexdump (pyproject.toml): finished with status 'done'
  Created wheel for hexdump: filename=hexdump-3.3-py3-none-any.whl size=8932 sha256=885bab21893ec8bbac84cbb13fa1c0f48ac3c349eff6657d2d272b317af69d8a
  Stored in directory: /root/.cache/pip/wheels/5e/b9/b5/227a20e7e8bbdb2b17e46a087c0f0119059ee65fe8374cac18
  Building wheel for pyaes (pyproject.toml): started
  Building wheel for pyaes (pyproject.toml): finished with status 'done'
  Created wheel for pyaes: filename=pyaes-1.6.1-py3-none-any.whl size=26422 sha256=f4221ef8ae0b377f1cefff9285e4d741b619258750e8f807cde266fef496a28d
  Stored in directory: /root/.cache/pip/wheels/d9/43/32/ec313dd557e88e419489b9df40c04dad7b99576bde162496f8
Successfully built hexdump pyaes
Installing collected packages: xmodem, pyserial, pyaes, hexdump, zipp, wcwidth, urllib3, semantic-version, py-datastruct, ordered-set, idna, colorama, click, charset_normalizer, certifi, bitstruct, ymodem, requests, prettytable, importlib-metadata, bk7231tools, ltchiptool
Successfully installed bitstruct-8.21.0 bk7231tools-2.1.0 certifi-2025.8.3 charset_normalizer-3.4.3 click-8.3.0 colorama-0.4.6 hexdump-3.3 idna-3.10 importlib-metadata-8.7.0 ltchiptool-4.12.2 ordered-set-4.1.0 prettytable-3.16.0 py-datastruct-1.1.0 pyaes-1.6.1 pyserial-3.5 requests-2.32.5 semantic-version-2.10.0 urllib3-2.5.0 wcwidth-0.2.14 xmodem-0.4.7 ymodem-1.5.1 zipp-3.23.0
Warning! Non-Git installations are NOT SUPPORTED.
HARDWARE: BK7231N 120MHz, 256KB RAM, 1.03MB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a 
 - framework-beken-bdk @ 0.0.0+v2021.06.07.sha.6491b8c 
 - library-flashdb @ 1.2.0+sha.d5c892f 
 - library-freertos @ 9.0.0+sha.ff31b7d 
 - library-freertos-port @ 2025.1.18+sha.dbfc408 
 - library-lwip @ 2.2.0-bdk+sha.48da4a2 
 - library-printf @ 6.1.0+sha.28a79bd 
 - library-uf2ota @ 5.0.0+sha.f955412 
PLATFORM VERSIONS:
 - libretiny @ 1.9.2
 - ltchiptool @ 4.12.2
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.9.1
Dependency Graph
|-- AsyncTCP @ 3.4.5
|-- ESPAsyncWebServer @ 3.7.10
|-- noise-c @ 0.1.10
|-- ArduinoJson @ 7.4.2
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_frame_helper_noise.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/esphome/ota/ota_esphome.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/json/json_util.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/libretiny/core.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/libretiny/gpio_arduino.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/libretiny/helpers.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/libretiny/lt_component.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/libretiny/preferences.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/logger/logger_libretiny.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/mdns/mdns_libretiny.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/ota/ota_backend_arduino_libretiny.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/select/select.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/select/select_call.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/select/select_traits.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/sensor/sensor.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/socket/lwip_sockets_impl.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/socket/socket.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/switch/automation.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/switch/switch.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/template/select/template_select.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/template/sensor/template_sensor.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/template/switch/template_switch.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/uart/switch/uart_switch.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/uart/uart.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/uart/uart_component.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/uart/uart_component_libretiny.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/uart/uart_debugger.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/web_server/list_entities.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/web_server/web_server.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/components/wifi/wifi_component_libretiny.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/application.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/color.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/component.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/controller.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/log.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/string_ref.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/time.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/esphome/core/util.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/src/main.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/lib9f5/AsyncTCP/AsyncTCP.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncEventSource.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncJson.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncMessagePack.cpp.o
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncWebHeader.cpp.o
Archiving .pioenvs/dreo_dr-htf004s/lib9f5/libAsyncTCP.a
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncWebServerRequest.cpp.o
/config/esphome/dreo-space-heater.yaml: In lambda function:
/config/esphome/dreo-space-heater.yaml:495:72: error: no matching function for call to 'to_string(std::__cxx11::basic_string<char>&)'
  495 |           id(fan_mode).publish_state(to_string(id(fan_mode_memory_global)).c_str());
      |                                                                        ^
In file included from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/string:55,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/locale_classes.h:40,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ios_base.h:41,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/streambuf:41,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/streambuf_iterator.h:35,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/iterator:66,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ranges_algobase.h:36,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/ranges_algo.h:35,
                 from /data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/algorithm:64,
                 from /data/cache/platformio/platforms/libretiny/cores/common/arduino/src/Arduino.h:14,
                 from <command-line>:
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6688:3: note: candidate: 'std::string std::__cxx11::to_string(long double)'
 6688 |   to_string(long double __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6688:25: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long double'
 6688 |   to_string(long double __val)
      |             ~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6679:3: note: candidate: 'std::string std::__cxx11::to_string(double)'
 6679 |   to_string(double __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6679:20: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'double'
 6679 |   to_string(double __val)
      |             ~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6670:3: note: candidate: 'std::string std::__cxx11::to_string(float)'
 6670 |   to_string(float __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6670:19: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'float'
 6670 |   to_string(float __val)
      |             ~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6659:3: note: candidate: 'std::string std::__cxx11::to_string(long long unsigned int)'
 6659 |   to_string(unsigned long long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6659:32: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long long unsigned int'
 6659 |   to_string(unsigned long long __val)
      |             ~~~~~~~~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6647:3: note: candidate: 'std::string std::__cxx11::to_string(long long int)'
 6647 |   to_string(long long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6647:23: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long long int'
 6647 |   to_string(long long __val)
      |             ~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6639:3: note: candidate: 'std::string std::__cxx11::to_string(long unsigned int)'
 6639 |   to_string(unsigned long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6639:27: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long unsigned int'
 6639 |   to_string(unsigned long __val)
      |             ~~~~~~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6628:3: note: candidate: 'std::string std::__cxx11::to_string(long int)'
 6628 |   to_string(long __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6628:18: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'long int'
 6628 |   to_string(long __val)
      |             ~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6620:3: note: candidate: 'std::string std::__cxx11::to_string(unsigned int)'
 6620 |   to_string(unsigned __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6620:22: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'unsigned int'
 6620 |   to_string(unsigned __val)
      |             ~~~~~~~~~^~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6609:3: note: candidate: 'std::string std::__cxx11::to_string(int)'
 6609 |   to_string(int __val)
      |   ^~~~~~~~~
/data/cache/platformio/packages/toolchain-gccarmnoneeabi/arm-none-eabi/include/c++/10.3.1/bits/basic_string.h:6609:17: note:   no known conversion for argument 1 from 'std::__cxx11::basic_string<char>' to 'int'
 6609 |   to_string(int __val)
      |             ~~~~^~~~~
Compiling .pioenvs/dreo_dr-htf004s/lib16a/ESPAsyncWebServer/AsyncWebSocket.cpp.o
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:154: warning: "htons" redefined
  154 | #define htons(x) __htons(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo_dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:119: note: this is the location of the previous definition
  119 | #define htons(x) lwip_htons(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:155: warning: "ntohs" redefined
  155 | #define ntohs(x) __ntohs(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo_dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:120: note: this is the location of the previous definition
  120 | #define ntohs(x) lwip_ntohs(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:156: warning: "htonl" redefined
  156 | #define htonl(x) __htonl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo_dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:121: note: this is the location of the previous definition
  121 | #define htonl(x) lwip_htonl(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:157: warning: "ntohl" redefined
  157 | #define ntohl(x) __ntohl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/dreo_dr-htf004s/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/dreo_dr-htf004s/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:122: note: this is the location of the previous definition
  122 | #define ntohl(x) lwip_ntohl(x)
      | 
*** [.pioenvs/dreo_dr-htf004s/src/main.cpp.o] Error 1
========================= [FAILED] Took 383.40 seconds =========================
```

--- a/code/Dreo_DR-HTF004S.yaml
+++ b/code/Dreo_DR-HTF004S.yaml
@@ -502,7 +502,7 @@ switch:
     turn_on_action:
       - lambda: |-
           // Turn on and restore fan mode from memory before it was turned OFF
-          id(fan_mode).publish_state(to_string(id(fan_mode_memory_global)).c_str());
+          id(fan_mode).publish_state(id(fan_mode_memory_global).c_str());
     turn_off_action:
       - lambda: |-
           id(fan_mode).publish_state("OFF");

